cognite/neat/rules/transformers/_verification.py

```diff
@@ -0,0 +1,105 @@
+import warnings
+from abc import ABC
+from collections.abc import Iterator
+from contextlib import contextmanager
+from typing import Any, Literal
+
+from pydantic import ValidationError
+
+from cognite.neat.issues import IssueList, NeatError, NeatWarning
+from cognite.neat.rules._shared import T_InputRules, T_VerifiedRules
+from cognite.neat.rules.models import (
+    AssetRules,
+    AssetRulesInput,
+    DMSRules,
+    DMSRulesInput,
+    InformationRules,
+    InformationRulesInput,
+)
+
+from ._base import MaybeRules, OutRules, ReadRules, RulesTransformer
+
+
+class VerificationTransformer(RulesTransformer[T_InputRules, T_VerifiedRules], ABC):
+    """Base class for all verification transformers."""
+
+    _rules_cls: type[T_VerifiedRules]
+
+    def __init__(self, errors: Literal["raise", "continue"]) -> None:
+        self.errors = errors
+
+    def transform(self, rules: T_InputRules | OutRules[T_InputRules]) -> MaybeRules[T_VerifiedRules]:
+        issues = IssueList()
+        in_: T_InputRules = self._to_rules(rules)
+        error_args: dict[str, Any] = {}
+        if isinstance(rules, ReadRules):
+            error_args = rules.read_context
+        verified_rules: T_VerifiedRules | None = None
+        with _handle_issues(issues, NeatError, NeatWarning, error_args) as future:
+            verified_rules = self._rules_cls.model_validate(in_.dump())  # type: ignore[assignment]
```

Comment:
We moved _rules_cls all the validation methods inclluding post validation that runs on the entire class